src/UQpy/surrogates/polynomial_chaos/physics_informed/ConstrainedPCE.py

```diff
@@ -0,0 +1,322 @@
+
+        lar_path = regresion.lars_path(self.basis_extended, self.y_extended, max_iter=max_niter)[1]
+
+        steps = len(lar_path)
+
+        print('Obtained Steps: ', steps)
+        multindex = self.initial_pce.multi_index_set
+
+        if steps < 3:
+            raise Exception('LAR identified constant function! Check your data.')
+
+        best_err = np.inf
+        lar_basis = []
+        lar_index = []
+        lar_error = []
+
+        if virtual_niters == True and minsize_basis == 1:
+            minsize_basis = self.pde_data.nconst + 1
+
+        if minsize_basis > steps - 2 or no_iter == True:
+            minsize_basis = steps - 3
+
+        print('\nStart of the iterative LAR algorithm:')
+        print('-------------------------------------')
+        for i in range(minsize_basis, steps - 2):
+            print('\nStep No. ', i)
+            mask = lar_path[:i]
+            mask = np.concatenate([[0], mask])
+
+            multindex_step = multindex[mask, :]
+            basis_step = list(np.array(self.initial_pce.polynomial_basis.polynomials)[mask])
+
+            lar_index.append(multindex_step)
+            lar_basis.append(basis_step)
+
+            pce.polynomial_basis.polynomials_number = len(basis_step)
+            pce.polynomial_basis.polynomials = basis_step
+            pce.multi_index_set = multindex_step
+            pce.set_ed(pce.experimental_design_input, pce.experimental_design_output)
+
+            pce.coefficients = self.ols(pce, nvirtual=nvirtual)
+
+            err = self.estimate_error(pce, verif_S)
+            print('Error: ', err)
+            lar_error.append(err)
+
+            if err < best_err:
+                best_err = err
+                best_basis = basis_step
+                best_index = multindex_step
+
+            if best_err < target_error:
+                break
+
+        print('End of the iterative LAR algorithm:')
+        print('-------------------------------------')
+
+        print('\nLowest error: ', best_err)
+
+        if len(lar_error) > 1:
+            pce.polynomial_basis.polynomials_number = len(best_basis)
+            pce.polynomial_basis.polynomials = best_basis
+            pce.multi_index_set = best_index
+            pce.set_ed(pce.experimental_design_input, pce.experimental_design_output)
+            pce.coefficients = self.ols(pce, nvirtual=nvirtual)
+            err = self.estimate_error(pce, verif_S)
+        print('\nFinal PCE error: ', err)
+
+
+        self.lar_pce = pce
+        self.lar_basis = best_basis
+        self.lar_multindex = best_index
+        self.lar_error = best_err
+        self.lar_error_path = lar_error
+
+    def ols(self, pce=None, nvirtual=None, calc_coeff=True, return_coeff=True, n_PI=100):
+        """
+        Fit the sparse physics-informed PCE by ordinary least squares from Karush-Kuhn-Tucker normal equations
+
+        :param pce: an object of the :py:meth:`UQpy` :class:`PolynomialChaosExpansion` class
```

Comment:
For consistentence in formatting, please replace " :py:meth:`UQpy` :class:`PolynomialChaosExpansion`" with ":class:`.PolynomialChaosExpansion`" so the documentation will find the appropriate class